### PR TITLE
Convert table tests to subtests

### PR DESCRIPTION
**Pull request by zbintliff**
Addresses some of the functions in #175

### CODE REVIEW

**Comment by zbintliff** [comment]
To illustrate how they look with `-v`:

```
$ go test -v github.com/sdboyer/gps -run TestWorkmapToReach
=== RUN   TestWorkmapToReach
=== RUN   TestWorkmapToReach/wmToReach("transitive_err'd_package_no_backprop")
=== RUN   TestWorkmapToReach/wmToReach("internal_diamond")
=== RUN   TestWorkmapToReach/wmToReach("single")
=== RUN   TestWorkmapToReach/wmToReach("no_external_with_subpkg")
=== RUN   TestWorkmapToReach/wmToReach("missing_package_is_poison")
=== RUN   TestWorkmapToReach/wmToReach("transitive_missing_package_is_poison")
=== RUN   TestWorkmapToReach/wmToReach("rootmost_gets_imported")
=== RUN   TestWorkmapToReach/wmToReach("no_external")
=== RUN   TestWorkmapToReach/wmToReach("simple_base_transitive")
=== RUN   TestWorkmapToReach/wmToReach("err'd_package_is_poison")
=== RUN   TestWorkmapToReach/wmToReach("transitive_err'd_package_is_poison")
--- PASS: TestWorkmapToReach (0.00s)
    --- PASS: TestWorkmapToReach/wmToReach("transitive_err'd_package_no_backprop") (0.00s)
    --- PASS: TestWorkmapToReach/wmToReach("internal_diamond") (0.00s)
    --- PASS: TestWorkmapToReach/wmToReach("single") (0.00s)
    --- PASS: TestWorkmapToReach/wmToReach("no_external_with_subpkg") (0.00s)
    --- PASS: TestWorkmapToReach/wmToReach("missing_package_is_poison") (0.00s)
    --- PASS: TestWorkmapToReach/wmToReach("transitive_missing_package_is_poison") (0.00s)
    --- PASS: TestWorkmapToReach/wmToReach("rootmost_gets_imported") (0.00s)
    --- PASS: TestWorkmapToReach/wmToReach("no_external") (0.00s)
    --- PASS: TestWorkmapToReach/wmToReach("simple_base_transitive") (0.00s)
    --- PASS: TestWorkmapToReach/wmToReach("err'd_package_is_poison") (0.00s)
    --- PASS: TestWorkmapToReach/wmToReach("transitive_err'd_package_is_poison") (0.00s)
PASS
ok      github.com/sdboyer/gps  0.013s
```

**Comment by zbintliff** [comment]
Seems like cirlceci is afailing because it isn't using go 1.7 (no t.Run() func on testing)

File: analysis_test.go [hunk 1]
```diff
@@ -434,19 +434,21 @@ func TestWorkmapToReach(t *testing.T) {
 	for name, fix := range table {
 		// Avoid erroneous errors by initializing the fixture's error map if
 		// needed
-		if fix.em == nil {
-			fix.em = make(map[string]*ProblemImportError)
-		}
+		t.Run(fmt.Sprintf("wmToReach(%q)", name), func(t *testing.T) {
```
Comment (sdboyer) [inline analysis_test.go]: I think it's fine to just use the fixture name - it was only useful for providing context when the testing system itself didn't provide it directly. With subtests, that context is present.

File: deduce_test.go [hunk 2]
```diff
@@ -467,98 +467,102 @@ var pathDeductionFixtures = map[string][]pathDeductionFixture{
...
-				return buf.String()
-			case maybeGitSource:
-				return fmt.Sprintf("%T: %s", tmb, ufmt(tmb.url))
-			case maybeBzrSource:
-				return fmt.Sprintf("%T: %s", tmb, ufmt(tmb.url))
-			case maybeHgSource:
-				return fmt.Sprintf("%T: %s", tmb, ufmt(tmb.url))
-			case maybeGopkginSource:
-				return fmt.Sprintf("%T: %s (v%v) %s ", tmb, tmb.opath, tmb.major, ufmt(tmb.url))
+		t.Run(fmt.Sprintf("%s", typ), func(t *testing.T) {
```
Comment (sdboyer) [inline deduce_test.go]: Similar to above, just pass the string var directly

**Comment by sdboyer** [comment]
uggghhh. yes, that's why, i'd forgotten about that.

i've been meaning to switch off of circle to travis, anyway. i'll do that tonight (i hope), and you can rebase on master with the change.

**Comment by zbintliff** [comment]
@sdboyer while you can't comment on the diff if you add `?w=1` it will remove whitespace diffs.

**Comment by zbintliff** [comment]
I also had a question about the go routine in one of the tests. My understanding is that every call to `t.Run` is basically a new routine, so I wonder if we can remove that one call.

**Comment by zbintliff** [comment]
@sdboyer if you don't want to do the whole travis migration you can just upgrade the build image to Ubuntu 14.04.  It has [1.7.4](https://circleci.com/docs/1.0/language-go/) installed. The image currently being used only has 1.5.3

**Comment by sdboyer** [comment]
mm yeah, that is a quicker fix for now (since i didn't get to moving to travis last night). next commit you push should be built with 14.04

**Comment by zbintliff** [comment]
Hey @sdboyer I did a cursory look and seem to get the biggest table tests. When I run the tests individually they work. IT seems they are hitting a  10 minute timeout on circleci.

**Comment by sdboyer** [comment]
sorry for delay - i will likely merge later tonight, but when i started on the travis conversion, that led me to wanting to also just switch from glide to dep metadata files, which then led me to a bug in `dep init` (golang/dep#298) - got distracted to a few degrees of yak

**Comment by zbintliff** [comment]
It happens. No rush. I think there were a few cases where I removed
t.FailNow calls that I shouldn't have. So take you're time on it.

**Comment by sdboyer** [comment]
OK, tests passed locally, so merging. Thanks!

**Pull request merged**